common/src/main/java/org/opensearch/ml/common/connector/HttpConnector.java

```diff
@@ -324,13 +324,38 @@ public <T> T createPayload(String action, Map<String, String> parameters) {
             payload = fillNullParameters(parameters, payload);
             StringSubstitutor substitutor = new StringSubstitutor(parameters, "${parameters.", "}");
             payload = substitutor.replace(payload);
-
             if (!isJson(payload)) {
-                throw new IllegalArgumentException("Invalid payload: " + payload);
+                String payloadAfterEscape = connectorAction.get().getRequestBody();
+                Map<String, String> escapedParameters = escapeMapValues(parameters);
+                StringSubstitutor escapedSubstitutor = new StringSubstitutor(escapedParameters, "${parameters.", "}");
+                payloadAfterEscape = escapedSubstitutor.replace(payloadAfterEscape);
+                if (!isJson(payloadAfterEscape)) {
+                    throw new IllegalArgumentException("Invalid payload: " + payload);
+                } else {
+                    payload = payloadAfterEscape;
+                }
             }
             return (T) payload;
         }
         return (T) parameters.get("http_body");
+
+    }
+
+    public static Map<String, String> escapeMapValues(Map<String, String> parameters) {
+        Map<String, String> escapedMap = new HashMap<>();
+        if (parameters != null) {
+            for (Map.Entry<String, String> entry : parameters.entrySet()) {
+                String key = entry.getKey();
+                String value = entry.getValue();
+                String escapedValue = escapeValue(value);
+                escapedMap.put(key, escapedValue);
+            }
+        }
+        return escapedMap;
+    }
+
+    private static String escapeValue(String value) {
+        return value.replace("\\", "\\\\").replace("\"", "\\\"").replace("\n", "\\n").replace("\r", "\\r").replace("\t", "\\t");
```

Comment:
Why not use existing escape method? This escape method will replace "\n" in "This is a \new book" , which is not necessary. It may confuse user why same sentence will generate different embeddings after this PR